This is an optimization hack which should only be present
in a sparc 32bit driver of the compiler compiled with
host/target/build sparc64-redhat-linux --with-cpu=v7.
As long long HOST_WIDE_INT slows things down, we can have in
addition to the sparc64-*/2.96/{cc1,cc1plus}
sparc-*/2.96/{cc1,cc1plus} binaries which are suitable for compiling
-m32 code only, but use long HOST_WIDE_INT.

--- gcc/gcc.c.jj	Mon Jul 31 21:36:08 2000
+++ gcc/gcc.c	Mon Jul 31 21:44:42 2000
@@ -2870,6 +2870,8 @@ process_command (argc, argv)
   int have_c = 0;
   int have_o = 0;
   int lang_n_infiles = 0;
+  int m64 = 0;
+  int used_B = 0;
 
   GET_ENV_PATH_LIST (gcc_exec_prefix, "GCC_EXEC_PREFIX");
 
@@ -3203,6 +3205,7 @@ process_command (argc, argv)
 		spec_machine = p + 1;
 
 	      warn_std_ptr = &warn_std;
+	      used_B = 1;
 	      break;
 
 	    case 'B':
@@ -3246,6 +3249,7 @@ process_command (argc, argv)
 						       NULL_PTR),
 			    NULL_PTR,
 			    PREFIX_PRIORITY_B_OPT, 0, NULL_PTR);
+		used_B = 1;
                 n_switches++;
 	      }
 	      break;
@@ -3350,6 +3354,13 @@ process_command (argc, argv)
 #endif
 	      goto normal_switch;
 
+	    /* HACK START */
+	    case 'm':
+	      if (p[1] == '6' && p[2] == '4')
+		m64 = 1;
+	    /* FALLTHROUGH */
+	    /* HACK END */
+
 	    default:
 	    normal_switch:
 	      n_switches++;
@@ -3377,6 +3388,16 @@ process_command (argc, argv)
   /* Use 2 as fourth arg meaning try just the machine as a suffix,
      as well as trying the machine and the version.  */
 #ifndef OS2
+  /* HACK START */
+  if (!m64 && !used_B && !strncmp (spec_machine, "sparc64-", 8))
+    {
+      const char *sparc32_exec_prefix =
+	concat (standard_exec_prefix, "sparc-", spec_machine + 8,
+		dir_separator_str, spec_version, dir_separator_str, NULL_PTR);
+      add_prefix (&exec_prefixes, sparc32_exec_prefix, "GCC",
+		  PREFIX_PRIORITY_LAST, 0, warn_std_ptr);
+    }
+  /* HACK END */
   add_prefix (&exec_prefixes, standard_exec_prefix, "GCC",
 	      PREFIX_PRIORITY_LAST, 1, warn_std_ptr);
   add_prefix (&exec_prefixes, standard_exec_prefix, "BINUTILS",
